src/tasks/ivfm.py

```diff
@@ -0,0 +1,134 @@
+import os
+from functools import partial
+from pathlib import Path
+
+import numpy as np
+import SimpleITK as sitk
+import torch
+from einops.einops import rearrange
+from mat73 import loadmat
+
+from src.models.components.mlp import MLP
+from src.models.pinns_ivfm import PINNs_ivfm
+
+# Set device for training
+DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+# Seed for reproducibility
+np.random.seed(12345)
+torch.manual_seed(12345)
+if torch.cuda.is_available():
+    torch.cuda.manual_seed(12345)
+
+# Create data
+# Set current working directory
+cwd = Path(__file__).resolve().parent
+os.chdir(cwd)
+patient = "../../data/patient_06_1.mat"
+data = loadmat(patient)
+fname = os.path.basename(patient)[:-4]
+spacing = [0.5, 0.5, 1]
+
+# Create log folder
+save_dir = os.path.join("../../logs", "iVFM")
+os.makedirs(save_dir, exist_ok=True)
+
+# Weights for losses
+lambda_div = 1e-4
+lambda_BC = 1
+lambda_smooth = 1e-5
+
+nb_frame = data["uR"].shape[-1]
+preds_array = np.zeros([2, *data["uR"].shape])
+
+# nb_frame = 1
+
+# Create an MLP network
+mlp = MLP([2], [2], [100, 100, 100, 100, 100, 100, 100, 100], activation="tanh")
+
+# Define optimizer, learning rate, and number of epochs
+lr = 1e-4
+optimizer = partial(torch.optim.Adam, lr=lr)
+# optimizer = partial(torch.optim.LBFGS, lr=2, max_iter=100000)
```

Comment:
should we remove this comment line ?

Reply:
This is an optimizer that I want to try, since a lot of PINNs used the L-BFGS optimizer. However, it would require some code refactoring. To be discussed...

Reply:
OK